orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/securitygroup/SecurityGroupUpserter.groovy

```diff
@@ -33,7 +32,7 @@ interface SecurityGroupUpserter {
    * @return true when, according to the underlying cloud provider, the security group has been updated to match the
    * specified security group.
    */
-  boolean isSecurityGroupUpserted(MortService.SecurityGroup upsertedSecurityGroup, StageExecution stage)
+  boolean isSecurityGroupUpserted(com.netflix.spinnaker.orca.clouddriver.MortService.SecurityGroup upsertedSecurityGroup, StageExecution stage)
```

Comment:
I have absolutely no idea why, but without giving the fully qualified name this fails to compile. Doesn't show any errors in my IDE without the fully qualified name and I don't know enough about Groovy/Java transpiling to debug it.

Reply:
FWIW, this happens to me as well.  With the fully qualified name, IntelliJ warns about an "Unnecessary qualified reference", but then when I replace it with
```
diff --git a/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/securitygroup/SecurityGroupUpserter.groovy b/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/>
index 8d44739ef..d66a00c42 100644
--- a/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/securitygroup/SecurityGroupUpserter.groovy
+++ b/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/securitygroup/SecurityGroupUpserter.groovy
@@ -17,6 +17,7 @@
 package com.netflix.spinnaker.orca.clouddriver.tasks.securitygroup
 
 import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution
+import com.netflix.spinnaker.orca.clouddriver.MortService
 import groovy.transform.Canonical
 
 interface SecurityGroupUpserter {
@@ -32,7 +33,7 @@ interface SecurityGroupUpserter {
    * @return true when, according to the underlying cloud provider, the security group has been updated to match the
    * specified security group.
    */
-  boolean isSecurityGroupUpserted(com.netflix.spinnaker.orca.clouddriver.MortService.SecurityGroup upsertedSecurityGroup, StageExecution stage)
+  boolean isSecurityGroupUpserted(MortService.SecurityGroup upsertedSecurityGroup, StageExecution stage)
 
   /**
    * @return The cloud provider type that this object supports.
```
I get:
```
$ ./gradlew build

> Task :orca-clouddriver:compileGroovy
/Users/dbyron/src/spinnaker/orca/orca-clouddriver/build/tmp/compileGroovy/groovy-java-stubs/com/netflix/spinnaker/orca/clouddriver/tasks/securitygroup/SecurityGroupUpserter.java:15: error: package MortService does not exist
 boolean isSecurityGroupUpserted(MortService.SecurityGroup upsertedSecurityGroup, com.netflix.spinnaker.orca.api.pipeline.models.StageExecution stage);
                                            ^
1 error
startup failed:
Compilation failed; see the compiler error output for details.

1 error


> Task :orca-clouddriver:compileGroovy FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':orca-clouddriver:compileGroovy'.
> Compilation failed; see the compiler error output for details.
```
Note that the compile error is in SecurityGroupUpserter.java -- a generated file.

My next move is to try converting SecurityGroupUpserter.groovy into java, which didn't turn out to be so bad....and works without the fully qualified name here.  If you like, I'll push a commit that does this.

Reply:
If you've got it working you're welcome to push it, or I can replicate this next week. Thanks!

Reply:
Pushed https://github.com/spinnaker/orca/pull/4269/commits/f39c69daa72a23dba7a64df55fcbbcb4fac22aa6.